Common/src/Common/ProtoSerializer.cs

```diff
@@ -157,65 +113,68 @@ public static void RegisterClass(Type classType)
                                       object obj,
                                       Type   type)
   {
-    var field = typeLookup.Single(pair => pair.Value == type)
-                          .Key;
+    var field = TypeLookup.IndexOf(type) + 1;
 
     Serializer.NonGeneric.SerializeWithLengthPrefix(stream,
                                                     obj,
                                                     PrefixStyle.Base128,
                                                     field);
   }
 
-  private static bool ReadNext(Stream     stream,
-                               out object obj)
+  private static bool ReadNext(Stream      stream,
+                               out object? obj)
   {
     if (!Serializer.NonGeneric.TryDeserializeWithLengthPrefix(stream,
                                                               PrefixStyle.Base128,
-                                                              field =>
-                                                              {
-                                                                return typeLookup[field];
-                                                              },
+                                                              field => TypeLookup[field - 1],
```

Comment:
```suggestion
                                                              field => TypeLookup[field - 1],  // "-1" to have 1-based indexing instead of 0-based indexing. Required by protobuf.
```